Richard/delete, update and edit using observables

kokobear3327 opened the pull request.

ahaly92 commented on src/app/admin/product-form/product-form.component.html:
```diff
@@ -89,14 +89,23 @@
         </div>
       </div>
       <button class="btn-primary">Save</button>
+      <br />
```
use styling over multiple <br/> tags

ahaly92 commented on src/app/admin/product-form/product-form.component.ts:
```diff
@@ -2,29 +2,49 @@ import { CategoryService } from './../../category.service';
+    if (this.id)
+      this.subscription = this.productService
+        .get(this.id)
+        .subscribe(product => (this.product = product));
```
this is simple enough for async pipe no?

kokobear3327 replied:
There is not place in the html to put the async pipe?  This is drawing the id from the product.

ahaly92 replied:
💯

ahaly92 commented on src/app/admin/product-form/product-form.component.ts:
```diff
@@ -2,29 +2,49 @@ import { CategoryService } from './../../category.service';
+    }
+  }
+
+  ngOnDestroy() {
```
if we go the async route lose the ngOnDestroy

ahaly92 commented on src/app/models/product.ts:
```diff
@@ -1,5 +1,5 @@
 export interface Product {
-  key: string;
+  $key: string;
```
key is more consistent with the the rest of the code. lose $

ahaly92 commented on src/app/product.service.ts:
```diff
@@ -19,4 +20,33 @@ export class ProductService {
+        map(changes => {
+          return changes.map(rawProduct => {
+            const data = rawProduct.payload.val();
+            const $key = rawProduct.payload.key;
```
change to key

ahaly92 commented on src/app/admin/admin-products/admin-products.component.ts:
```diff
@@ -1,7 +1,7 @@
 import { Observable } from 'rxjs';
 import { ProductService } from './../../product.service';
-import { Component } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
```
are you using this OnDestroy anymore? If not clean up imports

ahaly92 commented:
one small change but then :shipit:

Commit:
feature: editing, updating and deleting of products working with observables pattern






feature: editing, updating and deleting of products working with observables pattern


fix: with external firebaseUtilityFunction


fix: with external firebaseUtilityFunction


fix: with external firebaseUtilityFunction
